sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerBuilder.java

```diff
@@ -0,0 +1,503 @@
+        configValidators.addAll(DefaultConfigValidators.getValidators(schedulerConfig));
+        configValidators.addAll(customConfigValidators);
+        final ConfigurationUpdater.UpdateResult configUpdateResult =
+                updateConfig(serviceSpec, stateStore, configStore, configValidators);
+
+        // Now generate the new plans
+        Collection<Plan> plans = getPlans(stateStore, configStore, serviceSpec, manualPlans, yamlPlans);
+
+        if (!configUpdateResult.getErrors().isEmpty()) {
+            LOGGER.warn("Failed to update configuration due to errors with configuration {}: {}",
+                    configUpdateResult.getTargetId(), configUpdateResult.getErrors());
+            try {
+                // If there were errors, stick with the last accepted target configuration.
+                serviceSpec = configStore.fetch(configStore.getTargetConfig());
+            } catch (ConfigStoreException e) {
+                LOGGER.error("Failed to retrieve previous target configuration.");
+                throw new IllegalArgumentException(e);
+            }
+        }
+
+        plans = overrideDeployPlan(plans, lastCompletedDeploymentType);
+        Optional<Plan> deployOptional = SchedulerUtils.getDeployPlan(plans);
+        if (!deployOptional.isPresent()) {
+            throw new IllegalArgumentException("No deploy plan provided: " + plans);
+        }
+
+        List<String> errors = configUpdateResult.getErrors().stream()
+                .map(ConfigValidationError::toString)
+                .collect(Collectors.toList());
+        if (!errors.isEmpty()) {
+            plans = updateDeployPlan(plans, errors);
+        }
+
+        return new DefaultScheduler(
+                serviceSpec,
+                schedulerConfig,
+                customResources,
+                plans,
+                stateStore,
+                configStore,
+                endpointProducers,
+                Optional.ofNullable(recoveryPlanOverriderFactory));
+    }
+
+    /**
+     * Gets or generate plans against the given service spec.
+     *
+     * @param stateStore The state store to use for plan generation.
+     * @param configStore The config store to use for plan generation.
+     * @return a collection of plans
+     */
+    private static Collection<Plan> getPlans(
+            StateStore stateStore,
+            ConfigStore<ServiceSpec> configStore,
+            ServiceSpec serviceSpec,
+            List<Plan> manualPlans,
+            Map<String, RawPlan> yamlPlans) {
+        final String plansType;
+        final Collection<Plan> plans;
+        if (!manualPlans.isEmpty()) {
+            plansType = "manual";
+            plans = new ArrayList<>(manualPlans);
+        } else if (!yamlPlans.isEmpty()) {
+            plansType = "YAML";
+            // Note: Any internal Plan generation must only be AFTER updating/validating the config. Otherwise plans
```

Comment:
Is there a way to check *in code* whether this has been done?

Reply:
getPlans() is technically called twice, once before the config update is checked and once after. The before case is to determine if the deployment was complete according to the previous config. (see L278-L287)